### DISCUSSION TITLE
unstable_revalidate doesn't update the page on client side routing

### DISCUSSION THREAD

**Discussion by Hooman-studio** [vercel/next.js#37096] [Help]
On a website for a client I use the new on-demand revalidation feature by Nextjs to update the news page (and corresponding article) when one gets updated. The problem however is that on client side navigation I get the old site.

For example, go to [https://greencardbyinvestment.com/eb-5-news](https://greencardbyinvestment.com/eb-5-news) and see the current page. But if you go to [https://greencardbyinvestment.com/](https://greencardbyinvestment.com/) and then navigate to news, it shows and older version of the page. Is this fixed in the next nextjs update?

**Comment by icyJoseph** [comment]
I knew that the page remains the same under the same CSP session (so to speak), but refreshing the page, on any route really, should make it so that the app loads the latest data when you arrive at the revalidated route.

I tested this, with a very simple app, and it does seem to work, as expected. I navigate between home and blog routes, and see the same data, even though I have sent a revalidate (successful) request, when I refresh I see a new blog route, independently of how I navigate. Could there be some stale data in your app? Perhaps `getInitialProps` somewhere? Is it possible to take a look at the repository? Or could you try to make a reproduction demo?

**Comment by Hooman-studio** [comment]
I am using getInitialProps in my sitemap page, that is the only place I use it. Does this have influence on other pages as well? It is hard for me to publish the repo as it is for a client, and it would take a long time to reproduce since I don't know what the issue is/which parts I should copy over. I don't have any data fetching on the `_app.tsx` and `_document.tsx` files, on the page example I gave I am using the following data fetching method:

```
export const getStaticProps = async () => {
  const highlight = await prisma.article.findFirst({
    where: {
      status: 2,
    },
    select: {
      id: true,
      title: true,
      slug: true,
      image: true,
      imagedescription: true,
    },
  });

  const articles = await getArticles({ take: 20, skip: 0 });
  const count = await getCount({});
  const numberOfPages = count > 0 ? Math.abs(Math.floor((count - 1) / 20)) : 0;
  const pages = new Array(numberOfPages).fill(null).map((_, idx) => idx + 1);
  const categories = await getCategories();
  const currentPage = 1;

  return {
    props: {
      highlight,
      data: articles ?? [],
      categories,
      pages,
      currentPage,
    },
  };
};
```

**edit:**
The home page doesn't have any data fetching method (no `getStaticProps` or `getServerSideProps`, does that maybe have an influence?

**Reply by icyJoseph** [reply]
It should not have any impact. Now it seems to work? or perhaps the revalidate hasn't kicked in?

**Reply by Hooman-studio** [reply]
No I just updated/rebuild the site which set everything to the latest, now I changed the ordering again and it still has the same problem.
